### DISCUSSION TITLE
How to train a model on data containing missing values?

### DISCUSSION THREAD

**Discussion by miteshyadav** [awslabs/gluonts#1481] [Q&A]
## Description
I'm trying to run a DEEPAR model with multiple time series that contain a lot of missing/empty/NaN values.  Im refraining to impute missing values with 0, because that would be fallacious and the model wouldn't learn properly. Does GluonTS handle empty values by itself? Are there any other ideas or workaround this problem?

**Comment by lostella** [comment]
Could you include a minimal example of data that causes the issue?

In general, please include code examples that can be copy-pasted to reproduce the issue, otherwise it is hard for us to identify the problem.

**Comment by lostella** [comment]
Generally speaking though: If you include `None` in your “target” fields, DeepAR will handle them properly and treat them as missing.

**Reply by pratikgehlott** [reply]
What about values missing in the features, though we include `None` values in "target" fields, we need to also include `None or NaN` for the corresponding features, but after doing so it creates an error message stating gluonts cannot handle `NaN` values. Can you explain this scenario? 
@lostella @mbohlkeschneider 

Thanks

**Comment by mbohlkeschneider** [comment]
HI @pratikgehlott, 

This is correct. Missing values in features are currently not supported.

**Reply by pratikgehlott** [reply]
I need some suggestions, what would you do if you had a huge amount of missing values in a particular series. Because using imputing techniques would make the model learn wrong things.
@mbohlkeschneider 

Thanks

**Reply by mbohlkeschneider** [reply]
Depending on the amount and size of the missing value gap, it might be better to exclude the series from training. You can use this flag in `DeepAR` which will activate internal missing value handling. This is quite powerful - but of course has limits. You have to try on your use-case to see whether it works. https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/model/deepar/_estimator.py#L191

**Reply by pratikgehlott** [reply]
okay, thanks for the suggestion, will try to incorporate and experiment with this. 

Thanks
